src/gsi/cplr_gfs_ensmod.f90

```diff
@@ -440,31 +440,31 @@ subroutine move2bundle_(grd3d,en_loc3,atm_bundle,m_cvars2d,m_cvars3d,iret)
 !      if(trim(cvars2d(m))=='sst') sst=en_loc3(:,:,m_cvars2d(m)) !no sst for now
     enddo
 
-    km = en_perts(1,1)%grid%km
+    km1 = en_perts(1,1)%grid%km - 1
 !$omp parallel do  schedule(dynamic,1) private(m) 
     do m=1,nc3d
        if(trim(cvars3d(m))=='sf')then
-          u    = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          u    = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
        else if(trim(cvars3d(m))=='vp') then
-          v    = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          v    = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
        else if(trim(cvars3d(m))=='t')  then
-          tv   = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          tv   = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
        else if(trim(cvars3d(m))=='q')  then
-          q    = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          q    = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
        else if(trim(cvars3d(m))=='oz') then
-          oz   = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          oz   = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
        else if(trim(cvars3d(m))=='cw') then
-          cwmr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          cwmr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
        else if(trim(cvars3d(m))=='ql') then
-          qlmr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          qlmr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
        else if(trim(cvars3d(m))=='qi') then
-          qimr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          qimr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
        else if(trim(cvars3d(m))=='qr') then
-          qrmr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          qrmr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
        else if(trim(cvars3d(m))=='qs') then
-          qsmr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          qsmr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
        else if(trim(cvars3d(m))=='qg') then
-          qgmr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km)
+          qgmr = en_loc3(:,:,m_cvars3d(m):m_cvars3d(m)+km1)
```

Comment:
@RussTreadon-NOAA Does this bug fix change results?

Reply:
@emilyhcliu , interestingly specifying the proper upper bound on the array range does not alter cycled results.  

A hand waving argument is that the left hand side of the assignment statement is allocated to receive `km` elements.   Even though `km+1` locations are specified on the right hand side, only the first `km` are actually loaded into the left hand side.   We want the first `km` so the original and modified codes result in the same content being loaded into the left hand side.

Reply:
@RussTreadon-NOAA I understood now.  The original coding asked more (0, 1, ....km) than what we need (0,1, ... km-1).  Thanks.